### ISSUE TITLE
Resolution is lost when exporting high quality WCS layers

### ISSUE THREAD

**Issue by Samweli** [qgis/QGIS#45259]
### What is the bug or the crash?

When exporting a WCS layer with high resolution, the output file is exported with a different lower resolution.

screenshot

![wcs_layer_export](https://user-images.githubusercontent.com/2663775/135976402-2905feff-1c4b-4d35-8fa0-8c9f6cf0653a.gif)

### Steps to reproduce the issue

1. Add https://ows.digitalearth.africa/wcs as a WCS connection, connect to it and load layer with name 'Cropland extent 2019 - Eastern Africa'
2. Right click on the  layer and select export as -> save as
3. In the CRS input, select the layer CRS.
4. Click OK and wait for the result layer.

### Versions

QGIS 3.20

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by gioman** [comment]
> Right click on the layer and select export as -> save as

@Samweli this dialog does not also ask to choose the resolution?

**Comment by Samweli** [comment]
> @Samweli this dialog does not also ask to choose the resolution?

@gioman yes it does not, the option to select the layer resolution is disabled.

**Comment by gioman** [comment]
> @gioman yes it does not, the option to select the layer resolution is disabled.

See this (partial) saved at 10x10m

@Samweli but is there in your screencast the "resolution" field for horizontal and vertical resolution you can set to arbitrary values.
[crop1.zip](https://github.com/qgis/QGIS/files/7287901/crop1.zip)

**Comment by Samweli** [comment]
@gioman It would be nice if the user could be able to load the resolution from the layer itself.

**Comment by gioman** [comment]
@Samweli if you gdalinfo a WCS layer does it return the resolution?

Maybe this should be changed to feature request?

**Comment by Samweli** [comment]
> Maybe this should be changed to feature request?

yes @gioman
